Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Writer.hpp

```diff
@@ -49,6 +51,16 @@ class HepMC3Writer final : public WriterT<std::shared_ptr<HepMC3::GenEvent>> {
 
     /// The compression mode to use for the output file
     HepMC3Util::Compression compression = HepMC3Util::Compression::none;
+
+    /// Write events in order of their event number. If `false`, events are
+    /// written in whatever order they are processed.
+    bool writeEventsInOrder = true;
+
+    /// The maximum number of events to keep in the queue before writing
+    /// The value depends on the number of core and therefore the worst-case
+    /// distance between the *current* and the next event we can write to write
+    /// in order.
+    unsigned long maxEventsPending = 128;
```

Comment:
Nit: Perhaps a `/// NOTE` that this value doesn't mean anything if `writeEventsInOrder` is disabled? That's true, right?